lib/common/api/deprecate.js

```diff
@@ -73,7 +73,7 @@ const deprecate = {
     const warn = warnOnce(oldName, newName)
 
     if (typeof cb !== 'function') return promise
-    warn()
+    if (process.enablePromiseAPIs) warn()
```

Comment:
This should actually prevent a promise being returned at all.  If this flag is set to true, we should be returning undefined

Reply:
Resolving based on conversation in Slack ☝️